File: pset6/readability/readability.py
from cs50 import get_string

# Readability measure using Coleman-Liau index


# Get text input
def get_text():
    text = get_string("Text: ")
    return text


# Get word, letter and sentence count
def get_text_breakdown(text):
    word_count = 1;
    sentence_count = 0;
    letter_count = 0;
    for char in range(len(text)):
        c = text[char]
        if (c == " "):
            word_count = word_count + 1
        elif (c == '!' or c == '?' or c == '.'):
            sentence_count = sentence_count + 1
        else:
            letter_count = letter_count + 1
    counts_array = []
    counts_array.append(sentence_count)
    counts_array.append(letter_count)
    counts_array.append(word_count)
    return counts_array


# Get Coleman-Liau index value
# 0.0588 * L - 0.296 * S - 15.8,
# where L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text
def get_coleman_liau(counts_array):
    sentence_count = counts_array[0]
    letter_count = counts_array[1]
    word_count = counts_array[2]
    L = (letter_count / word_count) * 100
    S = (sentence_count / word_count) * 100
    coleman_liau_index_val = int(0.0588 * L - 0.296 * S - 15.8)
    if (coleman_liau_index_val < 1):
        return "Before Grade 1"
    elif (coleman_liau_index_val > 15):
        return "Grade 16+"
    else:
        return "Grade " + str(coleman_liau_index_val)

print(get_coleman_liau(get_text_breakdown(get_text())))

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

string get_text(string prompt);

// Request user's credit card, determine if checksum is met and if yes, test to see if Amex, Visa or Mastercard
int main(void)
{
    // Initializing word count to 1 as we will always miscount the final word as there is not space at end of sentence
    float word_count = 1, sentence_count = 0, letter_count = 0;
    string txt_str = get_string("Text: ");

    // Looping through each char in our string to identify letters, space (for words) and punctuation (!?.) (for sentences)
    for (int i = 0, len = strlen(txt_str); i < len; i++)
    {
        // Increment word count for any space found
        if (txt_str[i] == ' ')
        {
            word_count ++;
        }

        // Increment sentence count for any sentence ending punctuation found
        else if (txt_str[i] == '!' || txt_str[i] == '?' || txt_str[i] == '.')
        {
            sentence_count ++;
        }

        // Increment letter count for any alphabetic chars found
        else if 
        
        
        
        
        
        
        
        {
            letter_count ++;
        }
    }

    // Print letter, word and sentence counts
    printf("%.f letter(s)\n", letter_count);
    printf("%.f word(s)\n", word_count);
    printf("%.f sentence(s)\n", sentence_count);

    // Calculate letters per 100 words
    float let_100w = (letter_count / word_count) * 100;

    // Calculate sentences per 100 words
    float sen_100w = (sentence_count / word_count) * 100;

    // Calculate Coleman-Liau index
    float cl_index = 0.0588 * let_100w - 0.296 * sen_100w - 15.8;

    // Print grade level as determined by cl-index to screen
    if (cl_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (cl_index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.f\n", cl_index);
    }
}

// Prompt user to enter a text string
string get_text(string prompt)
{
    string s;
    s = get_string("%s", prompt);
    return s;
}